packages/jobs/lib/execution/sync.ts

```diff
@@ -167,6 +168,10 @@ export async function startSync(task: TaskSync, startScriptFn = startScript): Pr
         return Ok(nangoProps);
     } catch (err) {
         const error = new NangoError('sync_script_failure', { error: err instanceof Error ? err.message : err });
+        const syncJobId = syncJob?.id;
+        if (syncJobId) {
+            await updateSyncJobStatus(syncJobId, SyncStatus.STOPPED);
+        }
```

Comment:
move `updateSyncJobStatus`  out of the `onFailure` function to keep there only non critical logic